core/dbt/contracts/graph/nodes.py

```diff
@@ -1744,10 +1744,14 @@ def same_contents(self, old: Optional["SemanticModel"]) -> bool:
 
 
 @dataclass
-class SavedQuery(GraphNode):
+class SavedQueryMandatory(GraphNode):
```

Comment:
Yes we need it, because of the dataclass restriction about optional fields can't exists before required ones.
I followed a pattern we already have here.